File: dohiammk_manager_web/src/main/webapp/js/controller/goodsController.js
//控制层
app.controller('goodsController', function ($scope, $controller, itemCatService,goodsService) {

    $controller('baseController', {$scope: $scope});//继承

    //读取列表数据绑定到表单中  
    $scope.findAll = function () {
        goodsService.findAll().success(
            function (response) {
                $scope.list = response;
            }
        );
    }

    //分页
    $scope.findPage = function (page, rows) {
        goodsService.findPage(page, rows).success(
            function (response) {
                $scope.list = response.records;
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        );
    }

    //查询实体
    $scope.findOne = function (id) {
        goodsService.findOne(id).success(
            function (response) {
                $scope.entity = response;
            }
        );
    }

    //保存
    $scope.add = function () {
        goodsService.add($scope.entity).success(
            function (response) {
                if (response.success) {
                    alert("保存成功");
                    $scope.entity={}
                    //重新查询
                   // $scope.reloadList();//重新加载
                } else {
                    alert(response.message);
                }
            }
        );
    }


    //批量删除
    $scope.dele = function () {
        //获取选中的复选框
        goodsService.dele($scope.selectIds).success(
            function (response) {
                if (response.success) {
                    $scope.reloadList();//刷新列表
                    $scope.selectIds = [];
                }
            }
        );
    }

    $scope.searchEntity = {};//定义搜索对象

    //搜索
    $scope.search = function (page, rows) {
        goodsService.search(page, rows, $scope.searchEntity).success(
            function (response) {
                $scope.list = response.records;
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        );
    }


    //定义一个数组来判断商品审核的状态
    $scope.shopStatus = ['未申请', '申请中', '审核通过', '已驳回','取消申请'];
    //定义一个集合来存储分类内容
    $scope.mgItemList=[];
    $scope.findMyItemList=function () {
        itemCatService.findAll().success(
            function (response) {
                //alert("111");
                for(var i=0;i<response.length;i++){
                    $scope.mgItemList[response[i].id]=response[i].name;
                }
                //alert($scope.mgItemList[1])
            }
        )
    }

    $scope.auditOrRejectShop=function (status) {
        //获取选中的复选框
        goodsService.auditOrRejectShop($scope.selectIds,status).success(
            function (response) {
                if (response.success) {
                    $scope.reloadList();//刷新列表
                    $scope.selectIds = [];
                }
            }
        );
    }


});	

File: dohiammk_manager_web/src/main/java/com/pinyougou/manager/controller/GoodsController.java
package com.pinyougou.manager.controller;
import java.util.Arrays;
import java.util.List;
import com.alibaba.fastjson.JSON;
import com.pinyougou.page.service.ItemPageService;
import com.pinyougou.pojo.TbItem;
import com.pinyougou.pojogroup.Goods;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jms.core.JmsTemplate;
import org.springframework.jms.core.MessageCreator;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbGoods;
import com.pinyougou.sellergoods.service.GoodsService;
import entity.PageResult;
import entity.Result;
import javax.jms.*;

/**
 * controller
 * @author Administrator
 *
 */
@RestController
@RequestMapping("/goods")
public class GoodsController {

	@Reference
	private GoodsService goodsService;

	/*@Reference
	private SearchItemService searchItemService;*/

	@Reference
	private ItemPageService itemPageService;

	@Autowired
	private JmsTemplate jmsTemplate;

	@Autowired
	private Destination queueSolrDestination;//审核通过后添加索引库

	@Autowired
	private Destination queueSolrDelDestination;//根据goodsId删除
	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findAll")
	public List<TbGoods> findAll(){			
		return goodsService.findAll();
	}
	
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult  findPage(int page,int rows){			
		return goodsService.findPage(page, rows);
	}

	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findMyPage")
	public PageResult  findMyPage(@RequestBody TbGoods goods,int page,int rows){
		//System.out.println("加进来了吗？");
		return goodsService.findMyPage(goods,page, rows);
	}
	
	/**
	 * 增加
	 * @param goods
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody Goods goods){
		try {
			goodsService.add(goods);
			return new Result(true, "增加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "增加失败");
		}
	}
	
	/**
	 * 修改
	 * @param goods
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody TbGoods goods){
		try {
			goodsService.update(goods);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
	}	
	
	/**
	 * 获取实体
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public TbGoods findOne(Long id){
		return goodsService.findOne(id);		
	}
	
	/**
	 * 批量删除
	 * @param ids
	 * @return
	 */
	@RequestMapping("/delete")
	public Result delete(final Long [] ids){
		try {
			goodsService.delete(ids);
			jmsTemplate.send(queueSolrDelDestination, new MessageCreator() {
				@Override
				public Message createMessage(Session session) throws JMSException {

					return session.createObjectMessage(ids);
				}
			});
			//searchItemService.delGoods(Arrays.asList(ids));
			return new Result(true, "删除成功"); 
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "删除失败");
		}
	}
	
		/**
	 * 查询+分页
	 * @param goods
	 * @param page
	 * @param rows
	 * @return
	 */
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbGoods goods, int page, int rows  ){
		return goodsService.findPage(goods, page, rows);		
	}

	@RequestMapping("/auditOrRejectShop")
	public Result auditOrRejectShop(Long[] ids,String status){
		try {
			goodsService.commitShop(ids,status);
			System.out.println("状态："+status);
			if("2".equals(status)){
				final List<TbItem> tbItemList = goodsService.findItemByGoodIdAndStatus(ids, status);
                System.out.println("集合的长度"+tbItemList.size());
				final String toJSONString = JSON.toJSONString(tbItemList);
				if(tbItemList.size()>0){
					//通过消息中间件将修改的数据传递
					jmsTemplate.send(queueSolrDestination, new MessageCreator() {
						@Override
						public Message createMessage(Session session) throws JMSException {
							TextMessage message = session.createTextMessage(toJSONString);
							System.out.println("消息传递过去啦");
							return message;
						}
					});
					//searchItemService.saveList(tbItemList);
				}

			}
			return new Result(true, "审核或者驳回成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "审核或者驳回修改失败");
		}
	}

	//获取静态页面
	@RequestMapping("/getHtml")
	public void getHtml(Long goodsId){
		itemPageService.getHtml(goodsId);
	}
	
}
